pkg/repository/inmemory/filter_inmemory.go

```diff
@@ -1,54 +1,50 @@
 package inmemory
 
 import (
+	"context"
+	"encoding/hex"
 	"sync"
 
-	"github.com/btcsuite/btcd/chaincfg/chainhash"
-	"github.com/btcsuite/btcutil/gcs"
-	"github.com/btcsuite/btcutil/gcs/builder"
 	"github.com/vulpemventures/neutrino-elements/pkg/repository"
 )
 
+func makeUniqueKey(key repository.FilterKey) string {
+	return hex.EncodeToString(append(key.BlockHash, byte(key.FilterType)))
+}
```

Comment:
I think this could become a method of the FilterKey type and we can also improve it by making use of some hash function, like:
```
func (k FilterKey) String() string {
   hashedKey := btcutil.Hash160(append(k.BlockHash, byte(k.FilterType)
   return hex.EncodeToString(hashedKey[:6])
}
```

The advantage of moving this to pkg/repository is to not worry about reimplement it in every concrete repository, nor to worry about making sure that every implementation has exactly the same `makeUniqueKey()` function ;)